instance/models/instance.py

```diff
@@ -455,10 +455,8 @@ class AnsibleInstanceMixin(models.Model):
     """
     An instance that relies on Ansible to deploy its services
     """
-    ansible_source_repo_url = models.URLField(max_length=256,
-                                              default='https://github.com/edx/configuration.git')
-
-    configuration_version = models.CharField(max_length=50, default='master')
+    ansible_source_repo_url = models.URLField(max_length=256, blank=True)
+    configuration_version = models.CharField(max_length=50, blank=True)
```

Comment:
That's a very good observation. If we want to keep a provisioning history, we will need a separate model that will represent a single provisioning attempt/deployment of an `OpenEdxInstance`. It would store frozen configuration variables as they were at the time that particular provisioning was run. I would go further than simply storing `master`, if we wanted accurate history and repeatable builds, we will have to store the commit hash rather than the branch name. We would also have to freeze the version of each service that gets deployed to sandboxes (`cs_comments_service`, `notes_api`, `xqueue`, ...).

However since something like that is clearly out of scope of this PR, I suggest we do this for now:
- Load the default in a pre-save hook, similar to https://github.com/open-craft/opencraft/blob/545e0a2d43358f9ca0815e0aa9c19c62a5a144ce/instance/models/instance.py#L194-L197
- When reprovisioning an instance that was created from a github PR, check whether `edx_ansible_source_repo` and/or `configuration_version` are present, and use that, otherwise set `instance.ansible_source_repo` and `instance.configuration_version` to the defaults (even though they were not blank before)

This would make it possible to switch to a configuration fork when upstream is broken while not breaking existing instances that point to specific configuration repo/version.

Instances not created from PRs (although I'm not really sure we have any like that currently) would not be affected if defaults were changed. They would keep using stored `ansible_source_repo` and `configuration_version`. Existing instances created from PRs that had specific settings for these variables set in the PR body, would also not be affected. They would keep using the version/repo from PR. Existing PR instances without these settings would pick up the new defaults on each provisioning.